### ISSUE TITLE
allow task to override built-in commands (e.g `fmt`, `test`)

### ISSUE THREAD

**Issue by scarf005** [denoland/deno#27147]
- built-in commands like `deno test` suddenly get not so simple when permissions or external commands are required (e.g `deno test -RW` for snapshot tests)
- running `deno test` is better and less confusing `deno task test` for new contributors.

**Comment by bartlomieju** [comment]
Could you elaborate a bit more on the request here?

Do you want to have a way to customize how `deno test` works based on some config file option?

**Comment by dsherret** [comment]
I agree this is currently annoying, but I don't think we should do this. Instead I think we should improve the configuration file in other ways so that defining `deno task test` is not necessary.

**Comment by scarf005** [comment]
> Could you elaborate a bit more on the request here?
>
> Do you want to have a way to customize how `deno test` works based on some config file option?

another example: i need to use dprint on one of my project because i need to format lua files as well. in that case `deno fmt` is not useful; we use `deno task fmt` that invoke dprint.

**Comment by dsherret** [comment]
> another example: i need to use dprint on one of my project because i need to format lua files as well. in that case deno fmt is not useful; we use deno task fmt that invoke dprint.

For that scenario I think it would be more appropriate to have specific `fmt` configuration to do that. I don't think it should be done via `deno task`. For example, there needs to be a way to configure how file path arguments are provided to the formatter (probably needs to be streamed via stdin due to argument limits) and also perhaps how it works in the lsp.

**Comment by scarf005** [comment]
> For that scenario I think it would be more appropriate to have specific fmt configuration

would it mean that for such usecases, it'd be better as a (currently non-existent) "fmt" configuration in `deno.json`?

**Comment by scarf005** [comment]
also another legitimate usecases where overriding default test behavior is useful: 
```json
{
  "tasks": {
    "wasmbuild": "deno run -A jsr:@deno/wasmbuild",
    "test": {
        "dependencies": ["wasmbuild"],
        "command": "deno test"
    }
  }
}
```
user can easily forget to run `deno task test` over `deno test`. in the case of wasmbuild project, it's imperative the wasm build artifact is always synced before running any other task.
